checkstyle/suppressions.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0"?>
+<!DOCTYPE suppressions PUBLIC
+        "-//Puppy Crawl//DTD Suppressions 1.1//EN"
+        "http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
+
+<suppressions>
+
+  <!-- TODO -->
+  <suppress
+          checks="(MethodLength)"
+          files="(DataConverter|FieldsMetadata|JdbcSourceConnectorConfig).java"
+  />
+
+  <suppress
+          checks="(CyclomaticComplexity)"
+          files="(BufferedRecords|DataConverter|FieldsMetadata|JdbcSourceTask|PreparedStatementBinder|TimestampIncrementingTableQuerier|.*Dialect).java"
+  />
+
+  <suppress
+          checks="(ClassDataAbstractionCoupling)"
+          files="(DbDialect|JdbcSourceTask).java"
+  />
+
+  <suppress
+          checks="(JavaNCSS)"
+          files="(DataConverter|FieldsMetadata|JdbcSourceTask|TimestampIncrementingTableQuerier).java"
+  />
+
+  <suppress
+          checks="(NPathComplexity)"
+          files="(DataConverter|FieldsMetadata|JdbcSourceTask).java"
+  />
+
+  <suppress
```

Comment:
I think it's definitely possible we have a bug, but the lines listed here don't seem to make sense. The starting line corresponds to a block where we definitely have a `break` in the block (and the structure is *simpler* than earlier case statements in the file).

Are these lines just conservative? Are the errors the same even if the `lines` are set to start later?

Given the end position, I think this may have to do with this block of code: https://github.com/confluentinc/kafka-connect-jdbc/blob/master/src/main/java/io/confluent/connect/jdbc/source/DataConverter.java#L390-L414 The case for `Types.NUMERIC` is not guaranteed to exit the case statement since the entire block is dependent on `mapNumerics`, and if that is false it will fall through.

Perhaps this gets resolved by refactoring the fallthrough case and handling the `else` case for when `!mapNumerics` is `true`?